graylog2-web-interface/src/views/components/Query.jsx

```diff
@@ -56,7 +58,27 @@ const _renderWidgetGrid = (widgetDefs, widgetMapping, results, positions, queryI
   );
 };
 
+const _emptyDashboardInfo = () => {
+  return (
+    <Row className="content" style={{ marginRight: 0, marginLeft: 0 }}>
+      <Col md={12}>
+        <Jumbotron style={{ marginBottom: 0 }}>
+          <h2>This dashboard has no widgets yet</h2>
+          <p>
+            You can create a new widget by selecting a widget type in the left sidebar section &quot;Create&quot;.<br />
+            Have a look at the <DocumentationLink page={DocsHelper.PAGES.DASHBOARDS} text="documentation" />, to learn more about the widget creation.
+          </p>
+        </Jumbotron>
+      </Col>
+    </Row>
+  );
+};
+
 const Query = ({ allFields, fields, results, positions, widgetMapping, widgets, queryId }) => {
+  if (!widgets || widgets.size === 0) {
```

Comment:
Is there a more common way to check if an Immutable Map is empty?

And I tried to use == for the comparison, because in my opinion it makes totally sense, but there is an eslint to prevent this. Does this rule makes sense in this case?

Reply:
Another question I had, when do we use Immutable, is it a best practice for all "new data"

Reply:
Answer: Regarding immutable, we try to use it, if possible. The empty check is okay.

Regarding the comparison: we are always using `===`